addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js

```diff
@@ -0,0 +1,152 @@
+        );
+    }
+
+    getIgnoredColumns() {
+        return ["name"];
+    }
+
+    getActiveColumns(list) {
+        const activeColumns = super.getActiveColumns(list);
+        this.titleField = activeColumns.some((col) => col.name === "product_id")
+            ? "product_id"
+            : "name";
+        return activeColumns;
+    }
+}
+
+export class ProductLabelSectionAndNoteOne2Many extends X2ManyField {
+    static components = {
+        ...X2ManyField.components,
+        ListRenderer: ProductLabelSectionAndNoteListRender,
+    };
+}
+
+export const productLabelSectionAndNoteOne2Many = {
+    ...x2ManyField,
+    component: ProductLabelSectionAndNoteOne2Many,
+    additionalClasses: sectionAndNoteFieldOne2Many.additionalClasses,
+};
+
+registry
+    .category("fields")
+    .add("product_label_section_and_note_field_o2m", productLabelSectionAndNoteOne2Many);
+
+export class ProductLabelSectionAndNoteFieldAutocomplete extends Many2XAutocomplete {
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+    }
+
+    async loadOptionsSource(request) {
+        const options = await super.loadOptionsSource(request);
+        if (request.length) {
+            options.unshift({
+                label: _t('Label "%s"', request),
+                classList: "o_m2o_dropdown_option",
+                action: () => {
+                    this.props.record.update({ name: request });
+                },
+            });
+        }
+        return options;
+    }
+}
+
+export class ProductLabelSectionAndNoteField extends Many2OneField {
+    static components = {
+        ...Many2OneField.components,
+        Many2XAutocomplete: ProductLabelSectionAndNoteFieldAutocomplete,
+    };
+    static template = "account.ProductLabelSectionAndNoteField";
+
+    setup() {
+        super.setup();
+        this.extended = useState({ value: false });
+    }
+
+    get displayName() {
+        const { product_id, name } = this.props.record.data;
+        if (!product_id && !name) {
+            return false;
+        } else if (!product_id && name) {
+            return typeof name === "string" ? name.split("\n") : super.displayName;
+        } else {
+            return super.displayName;
+        }
+    }
+
+    get extraLines() {
+        const { product_id, name } = this.props.record.data;
+        if (name && ((product_id && this.extended.value) || !product_id)) {
+            return typeof name === "string" ? name.split("\n") : super.extraLines;
+        }
+        return false;
+    }
+
+    get hasProduct() {
+        const { product_id } = this.props.record.data;
+        return Boolean(product_id);
+    }
+
+    get Many2XAutocompleteProps() {
+        const props = super.Many2XAutocompleteProps;
+        props.record = this.props.record;
+        props.placeholder = _t("Chose a product/Write a label");
+        return props;
+    }
```

Comment:
I have a props validation error here. It seems props.value equals False and it couldn't because props validation is waiting for a String.
record is not a registered props on the Many2XAutocomplete class.